notebooks/Tutorial10_Geolocation_privacy_assessment.py

```diff
@@ -0,0 +1,251 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.15.2
+# ---
+
+# # Tutorial 10: Geolocation privacy assessment
+
+# In this tutorial, we demonstrate how to run a privacy assessment on geolocation trace data using an original dataset and a treated version of the same data.
+
+# ## Connection
+
+# +
+import os
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+url = os.environ.get("AVATAR_BASE_URL")
+username = os.environ.get("AVATAR_USERNAME")
+password = os.environ.get("AVATAR_PASSWORD")
+
+# +
+# This is the client that you'll be using for all of your requests
+from avatars.client import ApiClient
+from avatars.models import (
+    AvatarizationJobCreate,
+    AvatarizationParameters,
+    GeolocationDensityParameters,
+    GeolocationFeatures,
+    GeolocationFeaturesParameters,
+    JobStatus,
+    PointOfInterest,
+    PrivacyMetricsGeolocationJobCreate,
+    PrivacyMetricsGeolocationParameters,
+    PrivacyMetricsGeolocationScenario,
+    ReportGeolocationPrivacyCreate,
+    SignalPosition,
+)
+
+import pandas as pd
+import io
+
+# Change this to your actual server endpoint, e.g. base_url="https://avatar.company.com"
+client = ApiClient(base_url=url)
+client.authenticate(username=username, password=password)
+
+# +
+# Verify that we can connect to the API server
+client.health.get_health()
+
+# Verify that the client is compatible.
+client.compatibility.is_client_compatible()
+# -
+
+# ## Loading data
+
+# In this tutorial, we use the Porto taxis dataset, an open dataset often used to demonstrate geolocation data functionalities. The dataset is prepared to fit the format required by the geolocation privacy assessment feature, i.e. it contains the following columns: `id`, `t`, `lat` and `lon`.
+
+original_df = pd.read_csv("../fixtures/porto_taxi_200.csv")
+original_df
+
+# To demonstrate the privacy assessment, we will use two versions of the same data that have been altered in order to protect the privacy of the individuals, i.e. the drivers of the vehicles. Each version is treated following two distinct approaches:
+# - the first dataset (`trimmed_df`) is obtained by trimming / removing the start and end points of each trip.
+# - the second dataset (`noised_df`) is also obtained by trimming trips' start and end but noise is also added to each point.
+#
+# It is important to note that for the privacy assessment to be possible, the identifier (column `id`) need to be consistent across the datasets.
+
+trimmed_df = pd.read_csv("../fixtures/porto_taxi_trimmed_200.csv")
+trimmed_df
+
+noised_df = pd.read_csv("../fixtures/porto_taxi_noised_200.csv")
+noised_df
```

Comment:
Just for my information, which methods did you used to create noised and trimmed version of this dataset?

Reply:
I'll push it on another repo and share it with you

Reply:
Here: 
https://github.com/octopize/olivierrc-explorations/blob/main/taxi_geoloc_traces/Porto_traces_anonymizations.ipynb